catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptions.java

```diff
@@ -126,32 +95,35 @@ private void addOrUpdateMetacardsForRemoteRegistries(Set<String> sourceIds)
     }
 
     private Map<String, Metacard> getRegistryMetacardsMap() throws FederationAdminException {
-
-        Map<String, Metacard> registryMetacardsMap = new HashMap<>();
         try {
-            List<Metacard> registryMetacards = javax.security.auth.Subject.doAs(subject,
-                    (PrivilegedExceptionAction<List<Metacard>>) () -> federationAdminService.getRegistryMetacards());
+            List<Metacard> registryMetacards =
+                    Security.runAsAdminWithException(() -> federationAdminService.getRegistryMetacards());
 
-            for (Metacard metacard : registryMetacards) {
-                String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                registryMetacardsMap.put(registryId, metacard);
-            }
+            return registryMetacards.stream()
+                    .collect(Collectors.toMap(e -> e.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString(), Function.identity()));
         } catch (Exception e) {
             throw new FederationAdminException("Error querying for metacards ", e);
         }
-
-        return registryMetacardsMap;
     }
 
-    private Map<String, Metacard> getRemoteRegistryMetacardsMap(Set<String> sourceIds)
-            throws FederationAdminException {
+    private Map<String, Metacard> getRemoteRegistryMetacardsMap() throws FederationAdminException {
         Map<String, Metacard> remoteRegistryMetacards = new HashMap<>();
+        Set<String> sourceIds = registryStores.stream()
+                .filter(e -> e.isAvailable() && e.isPullAllowed())
```

Comment:
You really can write these as two separate `filter` calls. The stream processing will be optimized for you and you can make this cleaner and use method refs.